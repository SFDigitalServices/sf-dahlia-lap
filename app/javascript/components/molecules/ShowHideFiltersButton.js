import React from 'react'

import PropTypes from 'prop-types'

import Button from '../atoms/Button'
import StyledIcon from '../atoms/StyledIcon'

const getNumFiltersAppliedIcon = (numFiltersApplied) => {
  if (numFiltersApplied < 0 || numFiltersApplied > 9) {
    console.error(`Cannot display a number > 9 or < 0 in filter bubble: ${numFiltersApplied}`)
    numFiltersApplied = 9
  }

  return <StyledIcon icon={`filter-qty--${numFiltersApplied}`} />
}

const ShowHideFiltersButton = ({
  isShowingFilters = false,
  numFiltersApplied = 0,
  onClick = () => {}
}) => {
  const showFiltersCount = numFiltersApplied > 0
  return (
    <Button
      iconLeft={<StyledIcon icon='list-unordered' customSizeRem='1.25rem' />}
      iconRight={showFiltersCount ? getNumFiltersAppliedIcon(numFiltersApplied) : null}
      text={isShowingFilters ? 'Hide Filters' : 'Show Filters'}
      tightPadding
      // Set a min width so that there is no jitter when toggling between
      // show and hide text
      minWidthPx={showFiltersCount ? '236px' : '195px'}
      onClick={onClick}
    />
  )
}

ShowHideFiltersButton.propTypes = {
  isShowingFilters: PropTypes.bool,
  numFiltersApplied: PropTypes.number,
  onClick: PropTypes.func
}

export default ShowHideFiltersButton
