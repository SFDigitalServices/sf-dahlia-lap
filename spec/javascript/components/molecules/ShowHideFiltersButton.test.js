import React from 'react'

import { render, screen } from '@testing-library/react'

import ShowHideFiltersButton from 'components/molecules/ShowHideFiltersButton'

describe('ShowHideFiltersButton', () => {
  test('it renders properly with default props', () => {
    render(<ShowHideFiltersButton />)
    expect(screen.getByText('Show Filters')).toBeInTheDocument()
    expect(screen.queryByTestId('button-icon-right')).not.toBeInTheDocument()
    expect(screen.getByTestId('button-icon-left')).toBeInTheDocument()
  })

  test('it renders properly when isShowingFilters=true', () => {
    render(<ShowHideFiltersButton isShowingFilters />)
    expect(screen.getByText('Hide Filters')).toBeInTheDocument()
    expect(screen.queryByTestId('button-icon-right')).not.toBeInTheDocument()
    expect(screen.getByTestId('button-icon-left')).toBeInTheDocument()
  })

  test('it renders properly when numFiltersApplied=4', () => {
    const { asFragment } = render(<ShowHideFiltersButton isShowingFilters numFiltersApplied={4} />)
    expect(screen.getByTestId('button-icon-left')).toBeInTheDocument()
    expect(asFragment()).toMatchSnapshot()
  })
})
