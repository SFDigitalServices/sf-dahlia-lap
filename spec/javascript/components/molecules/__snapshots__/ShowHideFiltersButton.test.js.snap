// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ShowHideFiltersButton it renders properly when numFiltersApplied=4 1`] = `
<DocumentFragment>
  <button
    class="tight-padding"
    style="padding-top: 0px; padding-bottom: 0px; min-width: 236px;"
    type="button"
  >
    <div
      style="display: flex; align-items: center; justify-content: space-between; width: 100%;"
    >
      <span
        style="flex-shrink: 0; font-size: 0px;"
      >
        <svg
          data-testid="button-icon-left"
          style="width: 1.25rem; height: 1.25rem;"
        >
          <use
            xlink:href="#i-list-unordered"
          />
        </svg>
      </span>
      <span
        style="margin: 1rem 0.5rem 1rem 0.5rem; overflow: auto; flex-grow: 1; text-align: center;"
      >
        Hide Filters
      </span>
      <span
        style="flex-shrink: 0; font-size: 0px;"
      >
        <svg
          data-testid="button-icon-right"
          style="width: 1.5rem; height: 1.5rem;"
        >
          <use
            xlink:href="#i-filter-qty--4"
          />
        </svg>
      </span>
    </div>
  </button>
</DocumentFragment>
`;
